### PR TITLE
Migrate Portugal holidays to holiday groups

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

This migrates Portugal's holidays to the newer `holiday_groups` standard.

Perhaps these Christian holidays in Portugal could be converted to Holiday groups too?
- **St. Anthony Day** (JUN, 13)
- **Feast of Our Lady of Sorrows** (AUG, 20)
- **St. Matthew's Day** (SEP, 21)

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`, `make tox`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by PPsyrius** [comment]
Oof, pushed to wrong branch - fixed now.

**Review by arkid15r — COMMENTED** [review]
Hey @PPsyrius!
Great migration, the code looks clean. I had to do pretty much the same thing while working on https://github.com/dr-prodigy/python-holidays/pull/1057
I'm going to accept this anyway.

I think .po files can be omitted as no real l10n changes are present.

**Comment by arkid15r** [comment]
>Perhaps these Christian holidays in Portugal could be converted to Holiday groups too?

While working on holiday groups I added holidays that had been celebrated in at least 2 countries. If it's not something that is used in Portugal only feel free to add it to the group(s) too.

**Comment by PPsyrius** [comment]
> Hey @PPsyrius!
> Great migration, the code looks clean. I had to do pretty much the same thing while working on https://github.com/dr-prodigy/python-holidays/pull/1057
> I'm going to accept this anyway.
> 
> I think .po files can be omitted as no real l10n changes are present.

Will keep that in mind for future PRs. :)

File: holidays/locale/en_US/LC_MESSAGES/PT.po [hunk 1]
```diff
@@ -16,158 +16,158 @@ msgstr ""
 "Generated-By: pygettext.py 1.5\n"
 "X-Generator: Poedit 3.2.2\n"
 
-#: holidays/countries/portugal.py:83
+#: ./holidays/countries/portugal.py:86
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/locale/en_US/LC_MESSAGES/PT.po:19]: So I guess there is no easy way to do so?
Not sure if it's OS-specific behavior.

[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/locale/en_US/LC_MESSAGES/PT.po:19]: This is how paths are written by `polib` functions (on which `lingua pot-create` and our .po generation script are based). As a result, paths begin with "./" on Linux and ".\" on Windows.

[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/locale/en_US/LC_MESSAGES/PT.po:19]: Gotcha! Thanks for clarifying this.

**Pull request merged**